pkg/xds/context/mesh_context_builder.go

```diff
@@ -161,18 +161,25 @@ func (m *meshContextBuilder) BuildIfChanged(ctx context.Context, meshName string
 	}
 	meshServices := resources.MeshServices().Items
 	meshServicesByName := make(map[string]*v1alpha1.MeshServiceResource, len(meshServices))
+	meshServicesByLabels := map[string]map[string][]string{}
```

Comment:
It's not really `meshServicesByLabels` it's mor of `meshServiceNamesByLabelByValue`

I wonder. Do we need `map[string]map[string][]string{}` or could this be
```
meshServiceNamesByLabelValue := map[LabelValue]string
type LabelValue struct {
  Label string
  Value string
}

or maybe
`meshServicesByLabelValue := map[LabelValue]MeshServiceResource`
```